PE/__init__.py

```diff
@@ -833,21 +853,175 @@ def parseDelayImports(self):
+                        # can only be interpreted as an VA
+                        votes.append(False)
+
+                    elif (can_deref_as_rva, can_deref_as_va) == (False, False):
+                        # cannot be interpreted as either VA nor RVA
+                        pass
+
+                    elif (can_deref_as_rva, can_deref_as_va) == (True, True):
+                        # both VA and RVA interpretation could work.
+                        # so, we assume the difference between
+                        # the location of the pointer and the pointed-to location are fairly similar.
+                        # this is because the import table probably points to a nearby structure,
+                        # e.g., in the same section.
+                        #
+                        # so, is `abs(*ptr - ptr) < min rva`?
+                        # this should be the case for everything except:
+                        #  - REALLY large import tables, and
+                        #  - import tables split across more than one large section
+                        #
+                        # dereference as an RVA, and if the heuristic works, vote yes.
+                        # otherwise, its probably a VA.
+                        deref = self.readPointerAtRva(ptr)
+                        if abs(deref - ptr) < self.getMinRva():
+                            # the data is nearby, so the delta is small, so its probably an RVA.
+                            votes.append(True)
+                        else:
+                            # the treating the data as an RVA resulting in a large delta,
+                            # so its probably a VA.
+                            votes.append(False)
+                    else:
+                        # for clarity: all the cases are explicit above.
+                        raise Exception("impossible")
+
+                else:
+                    # its not an RVA,
+                    # and there' no overlap between valid VA and RVA ranges,
+                    # so it must be a VA.
+                    votes.append(False)
+
+            else:
+                # for clarity: all the cases are explicit above.
+                raise Exception("impossible")
+
+            irva += isize
+
+            if not self.checkRva(irva, size=isize):
+                # if import table is at the end of the file
+                # we may run to the end.
+                break
+
+            x.vsParse(self.readAtRva(irva, isize))
+
+        logger.debug("delayed imports: votes for RVAs: %d", sum([1 for vote in votes if vote]))
+        logger.debug("delayed imports: votes for VAs: %d", sum([1 for vote in votes if not vote]))
+
+        # more votes for RVA than for VA
+        return sum([1 for vote in votes if vote]) > sum([1 for vote in votes if not vote])
+
+    def parseImportTable(self, x, irva, flavor="import table", uses_rva=True):
```

Comment:
Is flavor ever going to expand beyond normal or delayed imports (I'm just curious about the change from a boolean to a string)?

Reply:
i dont know. `gcc delay import table`???

i found `is_import` to be confusing: if its not an import table, what is the alternative? the boolean wasn't documented anywhere. using the flavor string makes it very clear to the reader what variety of thing is being parsed.